src/metabase/automagic_dashboards/core.clj

```diff
@@ -44,23 +44,32 @@
 (def ^:private ^{:arglists '([field])} id-or-name
   (some-fn :id :name))
 
-(s/defn ->field
+(s/defn ->field :- (s/maybe (type Field))
   "Return `Field` instance for a given ID or name in the context of root."
-  [root id-or-name :- (s/cond-pre su/IntGreaterThanZero su/NonBlankString mbql.s/Field)]
-  (let [id-or-name (if (sequential? id-or-name)
-                     (filters/field-reference->id id-or-name)
-                     id-or-name)]
-    (if (->> root :source (instance? (type Table)))
-      (Field id-or-name)
-      (when-let [field (->> root
-                            :source
-                            :result_metadata
-                            (m/find-first (comp #{id-or-name} :name)))]
-        (-> field
-            (update :base_type keyword)
-            (update :semantic_type keyword)
-            field/map->FieldInstance
-            (classify/run-classifiers {}))))))
+  [{{result-metadata :result_metadata} :source, :as root}
+   field-id-or-name-or-clause :- (s/cond-pre su/IntGreaterThanZero su/NonBlankString mbql.s/Field)]
+  (let [id-or-name (if (sequential? field-id-or-name-or-clause)
+                     (filters/field-reference->id field-id-or-name-or-clause)
+                     field-id-or-name-or-clause)]
+    (or
+     ;; Handle integer Field IDs.
+     (when (integer? id-or-name)
+       (Field id-or-name))
+     ;; handle field string names. Only if we have result metadata. (Not sure why)
```

Comment:
I think the reason was underlying assumption that field string names are associated with the root entities that have results_metadata, sorta a duck type check